src/editor/action-executor.ts

```diff
@@ -140,13 +141,37 @@ export class ActionExecutor {
             }
 
             case EditActionType.DeleteNextToken: {
-                this.deleteCode(context.expressionToRight);
+                if (context.expressionToRight.rootNode instanceof ExprDotMethodStmt) {
+                    this.deleteCode(context.expressionToRight.rootNode);
+                } else this.deleteCode(context.expressionToRight);
 
                 break;
             }
 
             case EditActionType.DeletePrevToken: {
-                this.deleteCode(context.expressionToLeft);
+                if (context.expressionToLeft instanceof ExprDotMethodStmt) {
+                    // replace
+                    const initialBoundary = this.getBoundaries(context.expressionToLeft);
+                    const expr = context.expressionToLeft.getExpression();
+                    const insertionType = expr.canReplaceWithConstruct(expr);
+                    this.module.closeConstructDraftRecord(context.expressionToLeft);
+
+                    const root = context.expressionToLeft.getParentStatement();
+                    const dotMethodExpRoot = context.expressionToLeft.rootNode as Statement;
+                    context.expressionToLeft.onDelete();
```

Comment:
Do we want to add `onDelete()` as a callback in the constructor of `Statement` and call it through `construct.notify(CallbackType.delete)`?

Reply:
I added a github issue for this -> #259 it's still a work-in-progress polymorphism way of doing some pre-processings before deleting code constructs.